src/Traits/HasPermissions.php

```diff
@@ -7,20 +7,47 @@
 trait HasPermissions
 {
     /**
-     * Grant the given permission to a role.
+     * Grant the given permission(s) to a role.
      *
-     * @param  $permission
+     * @param string|array|Permission|\Illuminate\Support\Collection $permissions
+     * @param array $params
      *
      * @return HasPermissions
      */
-    public function givePermissionTo($permission)
+    public function givePermissionTo($permissions, ...$params)
     {
-        $this->permissions()->save($this->getStoredPermission($permission));
+        $joinings = $this->resolveJoiningAttributes($params);
```

Comment:
What does this do function do? Could you give an example? 

I'd rather see that `givePermissionTo` has only one parameter.